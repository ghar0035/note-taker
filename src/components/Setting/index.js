import { useState } from "react";
import axios from "axios";
import styled from "styled-components";
import settingImg from '../../images/setting.png'
import getCurrentUserName, { updateUserName } from "../../utilities/getCurrentUserName";
import getCurrentUserEmail from "../../utilities/getCurrentUserEmail";
import getCurrentUserId from "../../utilities/getCurrentUserId";



const Container = styled.div`
  width: 100%;
  display: flex;
  align-items: start;
`;

const Setting = () => {
  const [user, setUser] = useState({
    name: getCurrentUserName(),
  });

  const save = (e) => {
    e.preventDefault();

    axios.put('http://127.0.0.1:8001/user/' + getCurrentUserId(), user)
    .then(res => {
        console.log("nameSaved")
        updateUserName()
    } )
  }

  const onHandleChange = (e) => {
    const { name, value } = e.target;
    setUser((prevUser) => ({
      ...prevUser,
      [name]: value,
    }));
  };
  return (
    <Container>
      <form onSubmit={save}>
        <h4>Setting</h4>
        <div className="form-control">
          <label htmlFor="name">Name</label>
          <input
            id="name"
            name="name"
            type="text"
            onChange={onHandleChange}
            value={user.name}
          />
        </div>

        <p>
            <strong>Email:</strong> {getCurrentUserEmail()}
        </p>

        <button type="submit" className="btn">
          Save Changes
        </button>
      </form>
      <img  src={settingImg}  alt="" width="400" />
    </Container>
  );
};
export default Setting;
