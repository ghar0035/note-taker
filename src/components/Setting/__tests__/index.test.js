import renderer from "react-test-renderer";
import Setting from "../index";
import { MemoryRouter as Router } from "react-router-dom";

test("renders Setting", () => {
    const tree = renderer
      .create(
        <Router>
          <Setting />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });