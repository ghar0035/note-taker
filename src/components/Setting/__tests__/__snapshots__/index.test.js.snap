// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Setting 1`] = `
<div
  className="sc-bcXHqe grRzCf"
>
  <form
    onSubmit={[Function]}
  >
    <h4>
      Setting
    </h4>
    <div
      className="form-control"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <input
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
      />
    </div>
    <p>
      <strong>
        Email:
      </strong>
       
    </p>
    <button
      className="btn"
      type="submit"
    >
      Save
    </button>
  </form>
  <img
    alt=""
    src="setting.png"
    width="400"
  />
</div>
`;
